File: proconweb-api/src/main/java/br/com/procon/models/forms/AtendimentoForm.java
package br.com.procon.models.forms;

import java.io.Serializable;
import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

import javax.validation.constraints.NotNull;
import javax.validation.constraints.Size;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * 
 * @author Marlon Fernando Garcia
 *
 */
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class AtendimentoForm implements Serializable {

	private static final long serialVersionUID = 2498930558789511841L;
	
	private Integer id;
	@NotNull(message = "a data é obrigatória!")
	private LocalDate data;
	@Size(min = 1, message = "deve haver pelo menos um consumidor!")
	private List<Integer> consumidores = new ArrayList<>();
	@Size(min = 1, message = "deve haver pelo menos um fornecedor!")
	private List<Integer> fornecedores = new ArrayList<>();
	private String relato;
	@NotNull(message = "o atendente é obrigatório!")
	private Integer atendente;

}

File: proconweb/src/app/models/forms/atendimento-form.ts
import { Atendimento } from '../atendimento';

export class AtendimentoForm {
  id: number;
  data: string;
  consumidores: number[];
  fornecedores: number[];
  relato: string;
  atendente: number;

  constructor(atendimento: Atendimento) {
    this.id = atendimento.id;
    this.data = atendimento.data;
    this.consumidores = [];
    atendimento.consumidores.forEach((c) => this.consumidores.push(c.id));
    this.fornecedores = [];
    atendimento.fornecedores.forEach((f) => this.fornecedores.push(f.id));
    this.relato = atendimento.relato;
    this.atendente = atendimento.atendente.id;
  }
}
